### PR TITLE
Examples of pyomo benchmarks

### PR DESCRIPTION
**Pull request by teytaud**
## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Motivation and Context / Related issue

There are plenty of Pyomo models in the world.
Let me add some of them.
I start slowly with trivial models :-)

## Checklist

- [x] The documentation is up-to-date with the changes I made.
- [x] I have read the [**CONTRIBUTING**](https://facebookresearch.github.io/nevergrad/contributing.html) document and completed the CLA (see [**CLA**](https://facebookresearch.github.io/nevergrad/contributing.html#contributor-license-agreement-cla)).
- [x] All tests passed, and additional code has been covered with new tests.

### CODE REVIEW

**Comment by teytaud** [comment]
Do not merge for now. I want to add more meaningful Pyomo models first.

**Comment by teytaud** [comment]
Probably deprecated thanks to the better branch pyomoplusplus, will delete branch later unless something goes wrong there.

File: nevergrad/functions/pyomo/core.py [hunk 1]
```diff
@@ -139,7 +139,8 @@ def __init__(self, model: pyomo.Model) -> None:
         exp_tag = ",".join([n.name for n in self.all_objectives])
         exp_tag += "|" + ",".join([n.name for n in self.all_vars])
         exp_tag += "|" + ",".join([n.name for n in self.all_constraints])
-        self.register_initialization(name=exp_tag, model=self._model_instance)
+        self.register_initialization(model=self._model_instance)
```
Comment (teytaud) [inline nevergrad/functions/pyomo/core.py:142]: Presumably there was a bug here ? @herilalaina @jrapin

File: nevergrad/functions/pyomo/core.py [hunk 2]
```diff
@@ -73,6 +73,12 @@ def _make_pyomo_variable_to_parametrization(model_component: pyomo.Var, params:
                     params[params_name] = p.Choice(list(v.domain.ordered_data()))
                 else:
                     params[params_name] = p.Choice(list(v.domain.data()))
+            elif isinstance(v.domain, pyomo.RealSet) and not v.domains.bounds()[0] and not v.domains.bounds()[1]:
+                params[params_name] = p.Scalar()
+            elif isinstance(v.domain, pyomo.RealSet):
+                assert v.domain.bounds()[0], f"{k} ==> {v.domain.bounds()[0]}"
+                assert v.domain.bounds()[1], f"{k} ==> {v.domain.bounds()[1]}"
+                params[params_name] = p.Scalar(lower=v.domain.bounds()[0], upper=v.domain.bounds()[1])
```
Comment (teytaud) [inline nevergrad/functions/pyomo/core.py:81]: @herilalaina your opinion on this part is very welcome :-)

**Comment by teytaud** [comment]
A comment ( @herilalaina ) is that for the moment Nevergrad will not be competitive on white-box constraints, I'm on it -- this can be fixed but it's not yet implemented.

**Comment by teytaud** [comment]
I'm a bit surprised that pmedian with size 30 fails by time exceeded... this does not look THAT big.

**Comment by herilalaina** [comment]
Hey, I didn't work on pyomo. Maybe you want refer to @pacowong ? :)

**Comment by teytaud** [comment]
> Hey, I didn't work on pyomo. Maybe you want refer to @pacowong ? :)

oups sorry yes!

File: nevergrad/functions/pyomo/core.py (hunk 2 above)
Comment (teytaud) [reply nevergrad/functions/pyomo/core.py:81]: ok I remove it for now, seemingly there was an issue depending on the version of pyomo.

**Comment by teytaud** [comment]
I would merge this one, this is not a case in which we're good ==> so it's even more important to have it :-)

**Comment by pacowong** [comment]
I think the constraint checking procedure is slow. I can implement a parallel for loop in Python using [1] and use heuristics to decide which constraints to check first.

[1] https://joblib.readthedocs.io/en/latest/

**Comment by teytaud** [comment]
@pacowong  this PR looks reasonable to you ?

File: nevergrad/functions/pyomo/core.py [hunk 3]
```diff
@@ -175,3 +176,71 @@ def _pyomo_constraint_wrapper(self, i: int, instru: tp.ArgsKwargs) -> bool:
...
+    yield Pyomo(rosenbrock)
+
+    # Knapsack
+    for num_items in [4, 14, 44, 134]:
+        print(f"Creating Knapsack{num_items}")
+        if num_items == 4:
+            items = ['hammer', 'wrench', 'screwdriver', 'towel'] 
+            v = {'hammer':8, 'wrench':3, 'screwdriver':6, 'towel':11} 
+            w = {'hammer':5, 'wrench':7, 'screwdriver':4, 'towel':3} 
+            W_max = 14
```
Comment (jrapin) [inline nevergrad/functions/pyomo/core.py:204]: what are those variables, if you can, explicit names whould be helpful

**Comment by jrapin** [comment]
Also, please rename the PR with an explicit name

File: nevergrad/functions/pyomo/core.py (hunk 3 above)
Comment (teytaud) [reply nevergrad/functions/pyomo/core.py]: values and weights -- now fixed.

**Comment by pacowong** [comment]
> 
> 
> @pacowong this PR looks reasonable to you ?

Looks good.

**Comment by teytaud** [comment]
> I think the constraint checking procedure is slow. I can implement a parallel for loop in Python using [1] and use heuristics to decide which constraints to check first.
> 
> [1] https://joblib.readthedocs.io/en/latest/

Completely right, the constraint

**Pull request closed without merging**